services/userservice/userservice.go

```diff
@@ -26,6 +29,10 @@ func StoreNewUser(
 		hashedPassword,
 		creationTime,
 	); err != nil {
+		if err == databaseutil.UniqueConstraintError {
```

Comment:
Also this would be a good use of [error.Wrap](https://github.com/atmiguel/cerealnotes/pull/13/files#diff-09499008c3b5f7673ff66923b8758afaR32) <-- this is a new package btw, but i already included it in my PR.  Also i suspect that your throwing a singular error around, instead of creating a new one each time this situation happens happens. This means, on top of other issues,  that the stack trace here is not going to make sense.

Reply:
That link doesn't work for me

Reply:
this is the link i meant: https://godoc.org/github.com/pkg/errors#Wrap